distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Table_Helpers.enso

```diff
@@ -243,10 +242,10 @@ rename_columns internal_columns mapping on_problems =
                 _ ->
                     matched.add index
                     new_name = case Meta.type_of ms of
-                        Regex_Matcher.Regex_Matcher ->
+                        Regex_Matcher ->
                             pattern = ms.compile ((good_names.at index).at 0)
                             pattern.replace name ((good_names.at index).at 1)
-                        Text_Matcher.Text_Matcher -> (good_names.at index).at 1
+                        Text_Matcher -> (good_names.at index).at 1
```

Comment:
I know that this is not part of this PR, but wondering if we should be moving towards replacing `case Meta.type_of foo` with type-patterns (I think this pattern was there just because the latter were implemented later), i.e. I'd do `_ : Text_Matcher` instead and remove `Meta.type_of`.

I also wonder if there is any performance difference between the two. Seems like type pattern can be a bit faster than `Meta.type_of` which needs to materialize the metaobject. But I have no idea really. @JaroslavTulach @hubertp do you know if there should be a difference? Should we create a benchmark? I guess it may be worth knowing if we can use these two interchangeably or if there is a cost associated with one of them.

Reply:
literally only use - have removed.

Reply:
I see. Still may be interesting to understand the implications of each.